### ISSUE TITLE
add the note about using babel-plugin-lodash/assert in order to decrease `bundle` size

### ISSUE THREAD

**Issue by a-tarasyuk** [Microsoft/SimpleRestClients#42]
**SimpleRestClients** imports by default entire [`lodash`](https://github.com/Microsoft/SimpleRestClients/blob/master/src/SimpleWebRequest.ts#L9). Users who use **SimpleRestClients** and don't use _babel_ can have issues with `bundle` size. I've made an experiment with the [**optimized version**](https://github.com/a-tarasyuk/SimpleRestClients/tree/experiments) and the current production version - results provided below. 

Note: optimized version uses direct import - `import map from 'lodash/map'` and util function called `invariant` instead of entire `assert` library _(assert package provides many methods, however, usually library uses only one of them - `assert.ok.)`_. There is [unassert](https://www.npmjs.com/package/babel-plugin-unassert), however, this plugin completely removes asserts from [production code](https://www.npmjs.com/package/babel-plugin-unassert#example). 

Let's consider several use cases how users can use the library: 

1. A user uses the library without any optimizations and does not use lodash
    - [**optimized version**](https://github.com/a-tarasyuk/src-test/tree/without-any-optimizations-and-does-not-use-lodash)
    ```sh
          Asset      Size  Chunks             Chunk Names
      bundle.js  58.8 KiB       0  [emitted]  main
    ```
    - [**production version**](https://github.com/a-tarasyuk/src-test/tree/production-without-any-optimizations-and-does-not-use-lodash)
     ```sh
         Asset      Size  Chunks             Chunk Names
      bundle.js   108 KiB       0  [emitted]  main
    ```
2. A user uses the library with babel optimizations and does not use lodash

    - [**optimized version**](https://github.com/a-tarasyuk/src-test/tree/with-babel-optimizations-and-does-not-use-lodash)
    ```sh
          Asset      Size  Chunks             Chunk Names
      bundle.js  58.8 KiB       0  [emitted]  main
    ```
    - [**production version**](https://github.com/a-tarasyuk/src-test/tree/production-with-babel-optimizations-and-does-not-use-lodash)
     ```sh
          Asset      Size  Chunks             Chunk Names
      bundle.js   108 KiB       0  [emitted]  main
    ```
3. A user uses the library without any optimizations and uses lodash

    - [**optimized version**](https://github.com/a-tarasyuk/src-test/tree/without-any-optimizations-and-uses-lodash)
    ```sh
          Asset      Size  Chunks             Chunk Names
      bundle.js   128 KiB       0  [emitted]  main
    ```
    - [**production version**](https://github.com/a-tarasyuk/src-test/tree/production-without-any-optimizations-and-uses-lodash)
     ```sh
          Asset      Size  Chunks             Chunk Names
      bundle.js   108 KiB       0  [emitted]  main
    ```
4. A user uses the library with babel optimizations and uses lodash

    - [**optimized version**](https://github.com/a-tarasyuk/src-test/tree/with-babel-optimizations-and-uses-lodash)
    ```sh
          Asset      Size  Chunks             Chunk Names
      bundle.js    60 KiB       0  [emitted]  main
    ```
    - [**production version**](https://github.com/a-tarasyuk/src-test/tree/prodcution-with-babel-optimizations-and-uses-lodash)
     ```sh
          Asset      Size  Chunks             Chunk Names
      bundle.js   128 KiB       0  [emitted]  main
    ```
Seems that optimized has version only problems for the third case, and a user can solve it by adding babel optimization because she/he uses lodash in an application and have to do it. Would be very helpful to add an example of the optimal configuration for _webpack_/_rollup_/etc. Does that make sense?

**Comment by deregtd** [comment]
Yeah, feel free to submit a PR with an optimal configuration, or even modify the sample code to use it, and we can link it around from the various repos.
